### ISSUE TITLE
 no sound, randomly, viewing video with FF , Chrome, VLC

### ISSUE THREAD

**Issue by promeneur** [wwmm/easyeffects#1789]
### EasyEffects Version

6.2.6

### What package are you using?

openSUSE

### Distribution

openSUSE Tumbleweed

### Describe the bug

Randomly, sometime when viewing a video with FF or Chrome (YouTube) or VLC then there is no sound.

It seems this occurs after several starts and stops, or several launching and quitting.

Speakers are "on".

I must restart the PC to get again sound For FF or Chrome or VLC.

### Expected Behavior

Output sound is never cut.

### Debug Log

<details><summary>Debug Log</summary><br/>

```
Paste your log here
```
</details>

**Comment by wwmm** [comment]
This reminds of a bug whose source is still a mystery https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/1733. The next time this happens try to see if enabling and disabling the global bypass button, the one next to the presets menu, makes the sound come back. Or if disabling and enabling effects for the streams in the corresponding EasyEffects checkbox makes any difference. If one of these things work it is probably the same situation as the one in the issue I have opened in the PipeWire's page.

>I must restart the PC to get again sound For FF or Chrome or VLC.

Executing `easyeffects -q` and opening it again by clicking on you desktop icon is a faster solution if the global bypass procedure has no effect.

**Comment by promeneur** [comment]
pipewire 0.3.56

Ok, I will try your solution and I will report the result.

Thanks

**Comment by promeneur** [comment]
The problem has just occured with VLC. I get again the sound by clicking twice the "bypass" button.

**Comment by wwmm** [comment]
>The problem has just occured with VLC. I get again the sound by clicking twice the "bypass" button.

Ok. The next time this happens run `pw-dot` before forcing the sound to come back. The output file can be viewed  with the command `xdot`. It would be good to know if in your computer the links between filters are paused or if the sound is gone even if the links are made and active. In the past I saw paused links. But for a while I have been seeing lack of sound even if PipeWire tells that the links are made and active.

**Comment by promeneur** [comment]
Exactly what happens.

With FF I play a video.
After this video, I quit FF and launch VLC.
With VLC I play a video
After this video, I search and play another video.
From the beginning of the video, i get no sound. I pause the video.
I launch EE
I click twice the "bypass" button
I keep on playing the video. this time i get sound.

**Comment by LebedevRI** [comment]
I would just like to mention that this / https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/1733 is also happening for me,
during watching youtube videos in chrome on debian sid. No actionable reproduction steps, sadly.

**Comment by wwmm** [comment]
>No actionable reproduction steps, sadly.

That is one of the main problems with this bug. It is too random and as a result hard to debug. I think that just like https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/2617 we are facing some kind of bug in PipeWire. As far as PipeWire's API is concerned I do not see what else I could do different in order to avoid these 2 bugs.

**Comment by wwmm** [comment]
Whenever this "no sound" bug happens it is important to run `pw-dot` and maybe `pw-dump` too before forcing things to work again. This way we can see if anything new is happening to the link states.

**Comment by LebedevRI** [comment]
[pw-dump.txt](https://github.com/wwmm/easyeffects/files/9560727/pw-dump.txt)
[pw.dot.txt](https://github.com/wwmm/easyeffects/files/9560731/pw.dot.txt)

**Comment by LebedevRI** [comment]
Worse case: left channel is somehow random noise, right channel is good.
[pw-dump.txt](https://github.com/wwmm/easyeffects/files/9561233/pw-dump.txt)
[pw.dot.txt](https://github.com/wwmm/easyeffects/files/9561234/pw.dot.txt)

**Comment by wwmm** [comment]
>(left channel is silent)
>Worse case: left channel is somehow random noise, right channel is good.

The pw-dot output is totally fine. Just like I have seen in the last months when this problem happens. I will need some time to take a good look at the pw-dump file.

Such frustrating bug... According to PipeWire's own tool the links are there and in the active state. There should be audio but somehow there isn't.

**Comment by promeneur** [comment]
New phenomenon.

during a skype video talk suddenly no more mic.
before playing with "by pass" button, I executed "pw-dot".

[pw.dot.txt](https://github.com/wwmm/easyeffects/files/9564832/pw.dot.txt)

playing with "by pass" button does not solve the problem.

I recalled my contact then again after a while no more mic.

more info : the mic is dynamically shared between the host and a qemu guest.

I never get any problem till this day using mic with Skype.

**Comment by wwmm** [comment]
>before playing with "by pass" button, I executed "pw-dot".

And again the output is totally fine. The filters are loaded and the links between them are in the active state... I wonder what is happening inside PipeWire under the hoods. If the links are active they are literally sending silent audio buffers to the filters. It doesn't make any sense.

**Comment by promeneur** [comment]
More info about mic : I get an error message from Skype saying there is a problem with the mic and suggesting to restart the PC.

**Comment by promeneur** [comment]
I get the problem with VLC.

[pw.dot.txt](https://github.com/wwmm/easyeffects/files/9578654/pw.dot.txt)

Playing with "by pass" button does not solve the problem.

I had to relaunch VLC to get again sound.

**Comment by wwmm** [comment]
I wonder if PipeWire `0.3.58` will have any effect on this bug. At least another one whose source was unknown https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/2617 seems to have been unintentionally fixed by the new release.

**Comment by LebedevRI** [comment]
> I wonder if PipeWire `0.3.58` will have any effect on this bug.

I'll report back after statistically-significant time frame.

**Comment by LebedevRI** [comment]
> > I wonder if PipeWire `0.3.58` will have any effect on this bug.
> 
> I'll report back after statistically-significant time frame.

Just had the "left channel is somehow random noise, right channel is good." case again. So nothing changed basically.

**Comment by MrAureliusR** [comment]
I am having a somewhat similar issue; I didn't want to create a new issue in case this is the same bug.
I'm on Arch Linux, running KDE Plasma 5.25.5 using Pipewire 0.3.58 and Easyeffects 6.3.0-1. 
For ages, I've had no issues with EasyEffects. I use it both with the built-in speakers on my laptop, and with my Bluetooth JBL speaker. However, in the last few weeks (I should have noted exactly when but I didn't think of it) if EasyEffects is running, and I try to start an audio stream from _any_ application, the audio stream is unable to start and the application hangs _until_ I move the volume slider in the system tray. Then all of a sudden the program (YouTube in Firefox, VLC, Strawberry, whatever) will suddenly start playing and as long as the audio continues, it keeps playing fine. 
However, if the audio stops again for any reason (next track in a playlist, buffering, etc) the stream will once again get "hung" until I move the volume slider. The volume slider fix was something I discovered accidentally when trying to figure out what was going on. I don't see anything in dmesg. EasyEffects is installed as a system package from pacman, not as a Flatpak.
Looking in various log files, I can't immediately see anything that occurs at the same time as this problem, but I may not be looking in the right places. Checked journalctl for pipewire and pipewire-pulse. I occasionally see some things from easyeffects also but they seem to be related to GTK errors when drawing the interface so likely unrelated.

**Comment by wwmm** [comment]
>if EasyEffects is running, and I try to start an audio stream from any application, the audio stream is unable to start and the application hangs until I move the volume slider in the system tray.

If this is easy to reproduce you are probably seeing a different kind of problem. The bug in this issue is so random that it never happens when I need it to happen.

**Comment by promeneur** [comment]
After having a problem with the mic (see my previous comment 19 days before.) in EE settings, I disabled the input stream process.
Yesterday during a video talk with Skype again, no more mic. 
My conclusion is the problem does not come from EE, but comes from pipewire.  What do you think?

**Comment by wwmm** [comment]
>My conclusion is the problem does not come from EE, but comes from pipewire. What do you think?

It is the only explanation I can think of. In the issue I opened at the PipeWire's page I asked if some kind of special procedure was needed when creating/destroying links based on when the players streams enter/exit the playing state. The answer was that things should work out of the box. No delays or special procedures required. So as far as I can see I am doing nothing wrong.

**Comment by LebedevRI** [comment]
Still happening with pipewire 0.3.59, wireplumber 0.4.11, easyeffects 6.3.0.

**Comment by nargacu83** [comment]
I had the same problem and was using wireplumber. I tried to switch to pipewire-media-session, the problem still happened. 

The method i used to test was playing a video with Freetube (should be the same with firefox) and have pavucontrol opened on the side to see when the output of the app disappear and when it disappears unpause the video.

It seems like the microphone input works fine while this issue is happening, it appears to be only happening on the ouputs.

Replacing easyeffects with the Flatpak seems to have fixed the issue.

**Comment by promeneur** [comment]
So now what is the difference between the flatpack version and the openSUSE Tumbleweed version for example ?

**Comment by nargacu83** [comment]
> So now what is the difference between the flatpack version and the openSUSE Tumbleweed version for example ?

The flatpak version is using it's own dependencies and not the system ones.

I'm surprised to have this issue because i'm on Arch Linux like @wwmm and a friend a mine. Both of them doesn't seem to have the issue.

**Comment by promeneur** [comment]
>> The flatpak version is using it's own dependencies and not the system ones.

It's nos sufficient to say this. The question is where is the bug.

For example, openSUSE okular get a bug and not the flatpack version supplied by KDE team.
With their help we found that openSUSE okular uses a bad font which leads to a bug.
We must inquire with the help of EE team.

**Comment by wwmm** [comment]
>I'm surprised to have this issue because i'm on Arch Linux like @wwmm and a friend a mine. Both of them doesn't seem to have the issue.

This random lack of sound is super random and hard to reproduce. I still haven't found a way to make it happen when I want. It is always the same thing. After several hours without issue out of no where there is no sound after starting a new stream. I think that of all the software bugs I have ever seen in my life this one is the hardest to figure out. I have absolutely no idea about why it is even happening.

**Comment by nargacu83** [comment]
> It's nos sufficient to say this. The question is where is the bug.
> 
> For example, openSUSE okular get a bug and not the flatpack version supplied by KDE team. With their help we found that openSUSE okular uses a bad font which leads to a bug. We must inquire with the help of EE team.

Yeah sorry about that, i actually don't know what versions of packages it's using. I don't think it really matters because i actually encountered the same issue on the Flatpak version.

> This random lack of sound is super random and hard to reproduce. I still haven't found a way to make it happen when I want. It is always the same thing. After several hours without issue out of nowhere there is no sound after starting a new stream. I think that of all the software bugs I have ever seen in my life this one is the hardest to figure out. I have absolutely no idea about why it is even happening.

Agreed 100%, this is really annoying and hard to reproduce. Like you said, sometimes it will not happen at all for hours and sometimes it will happen every time you pause a stream.

I don't know if it's related, it seems to appear when i had the issue happening every time i did a quick journalctl command to check what's happening.  

This is the output for `journalctl --user -b -u pipewire-media-session`:

```
systemd[948]: Started PipeWire Multimedia Service.
pipewire[1031]: mod.rt: Can't find xdg-portal: (null)
pipewire[1031]: mod.rt: found session bus but no portal
pipewire[1031]: spa.audioadapter: 0x55c38a1d8518: scheduling stopped node
pipewire[1031]: spa.audioadapter: 0x55c38a1d8518: scheduling stopped node
pipewire[1031]: spa.audioadapter: 0x55c38a1d8518: scheduling stopped node
```

This is for `journalctl --user -b -u pipewire-media-session`:

```
systemd[948]: Started PipeWire Media Session Manager.
pipewire-media-session[1043]: mod.rt: Can't find xdg-portal: (null)
pipewire-media-session[1043]: mod.rt: found session bus but no portal
pipewire-media-session[1043]: dbus[1043]: Attempted to unregister path (path[0] = MediaEndpoint path[1] = A2DPSink) which isn't registered
pipewire-media-session[1043]: dbus[1043]: Attempted to unregister path (path[0] = MediaEndpoint path[1] = A2DPSink) which isn't registered
pipewire-media-session[1043]: dbus[1043]: Attempted to unregister path (path[0] = MediaEndpoint path[1] = A2DPSink) which isn't registered
etc...
```

**Comment by nargacu83** [comment]
Alright, i've been running easyeffects with this command: `G_MESSAGES_DEBUG=easyeffects easyeffects --gapplication-service`.

Here's the last few lines:
```
(easyeffects:25999): easyeffects-DEBUG: 14:53:31.907: 	pipe_manager.cpp:1189	Chromium port 110 is connected to easyeffects_sink port 58
(easyeffects:25999): easyeffects-DEBUG: 14:53:31.907: 	pipe_manager.cpp:1189	Chromium port 102 is connected to easyeffects_sink port 96
(easyeffects:25999): easyeffects-DEBUG: 14:53:31.908: 	stream_output_effects.cpp:156	At least one app linked to our device wants to play. Linking our filters.
(easyeffects:25999): easyeffects-DEBUG: 14:54:09.501: 	pipe_manager.cpp:212	Stream/Output/Audio 114 Chromium has been removed
```

When the issue happened, no new messages appeared. Only when the stream i paused was paused long enough to be removed. Started the stream again and it was working.

I was thinking it might be related on how easyeffects removes streams references or something?

**Comment by wwmm** [comment]
>I was thinking it might be related on how easyeffects removes streams references or something?

Not really. The one removing or adding streams is pipewire. Easyeffects is just listening to these events and linking/unlink filters accordingly. And these operations succeed. That is why I still think that this bug in on pipewire's side.

>I also ran the same command without --gapplication-service yesterday and i did not encounter the issue at all. Could it also be related to the running in the background thing?

Nothing different is done to the way we link/unlink filters when we run in background. We do not have a true service. It is just a gtk app with it's window hidden.

**Comment by DanMan** [comment]
Since YT videos don't even start playing, it might be that something is waiting for something else to happen, which is why there are basically no errors?

**Comment by wwmm** [comment]
>Since YT videos don't even start playing, it might be that something is waiting for something else to happen, which is why there are basically no errors?

Are you sure they do not start? At least in the occasions where I saw this bug happening on my computer  independently from the player used the stream was advancing like there was nothing wrong. IF it does not start you may be seeing a slightly different bug.

In this case if there are errors they are not going to be visible in EasyEffects logs. I would expect them to be in PipeWire's logs. Byt the way if we had a reliable way to reproduce the bug discussed in this issue PipeWire's logs would be the better place to investigate. But as I can't reproduce it when I need it I do not know what PipeWire is actually doing when it happens...

**Comment by wwmm** [comment]
>If someone needs a hotfix: https://github.com/Audio4Linux/JDSP4Linux/issues/74#issuecomment-1301701043

Interesting. I wonder if the built-in session manager is doing the same thing as wireplumber. I am almost sure I also saw this bug in the past while using the built-in session manager too.

**Comment by LebedevRI** [comment]
Still happening with pipewire 0.3.60-1, wireplumber 0.4.12-1+b1, easyeffects 7.0.0-1.

**Comment by HanabishiRecca** [comment]
> wireplumber 0.4.12

Just wait for the next release.

**Comment by promeneur** [comment]
Hello

For two months, I uninstalled EE.

Yesterday I got again the problem.
I was viewing a video with FF.
I paused the video, the time to get a coffee.
Furthermore, I played the video then no sound.
I mute/unmute the speakers at VLC icon (With KDE in the task bar in VLC icon, there is a mini-icon of a speaker for muting/unmuting sound). I also pause/play the video, then I get again the sound.

**Comment by wwmm** [comment]
>Yesterday I got again the problem.

For some reason I haven't seen this bug on my system for a few months. But based on one of the comments a PipeWire developer made in the issue I opened there this bug should still be present indeed. He does not think that the fix to wireplumber will solve this one :-(.

**Comment by Elendil211** [comment]
I have a similar problem that might be the same bug. It happens with Chromium and FreeTube. It happens very reliable so this might help?

Every time I have EasyEffects open, and then join a jitsi meeting, I have neither sound, nor video (can't talk, can't hear anything, can't start my camera). This happens in 100% of times. When I start the meeting first, and then open EasyEffects, everything works fine.

When I start a video on FreeTube and have EasyEffects running, in 100% of cases no video plays sound, and in 90% of cases no video plays. When I start EasyEffects after starting the video, everything works fine. 

The same video works fine in Chromium (also in Firefox).

It is not important whether or not I check the audio source in EasyEffects, or whether I start EasyEffects or Chromium/FreeTube first. 

FreeTube is Electron based, and shows as Chromium in EasyEffects, so it's very similar in that regard.

**Comment by wwmm** [comment]
>Every time I have EasyEffects open, and then join a jitsi meeting with Chromium, I have neither sound, nor video (can't talk, can't hear anything, can't start my camera).

I think you are the second person to report that videos do not start in some situations. This is something I have never seen in my computers and really sounds like a PipeWire problem. It does not make any sense something in EasyEffects would do this.

>It happens very reliable so this might help?

As you are able to reproduce this reliably my suggestion is trying to get PipeWire logs and reporting this whole `video not playing` issue to them. The issue discussed in this thread may or may not be related to how EasyEffects is doing things. But videos not playing is definitely beyond caused by some flaw on PipeWire's side.

**Comment by HanabishiRecca** [comment]
WirePlumber 0.4.13 released

> Fixes:
> 
>  - Worked around an issue that would prevent streams from properly linking
>    when using effects software like EasyEffects and JamesDSP

**Comment by DanMan** [comment]
> videos not playing is definitely caused by some flaw on PipeWire's side

Certainly possible, but I can reliably prevent all issues by simply not launching EE. So it appears to be something about how they interact.

**Comment by wwmm** [comment]
>Certainly possible, but I can reliably prevent all issues by simply not launching EE. So it appears to be something about how they interact.

There isn't much to be done. We just load the filters and link them on demand based on whether there are audio players active or not. Once the links are made everything is done by pipewire. We do not even have a way to pause the pipeline for example. It is all handled internally by pipewire.

Compared to not using easyeffects the main difference is that the audio player is linked to a null-sink. That we creat using a pipewire module. So again pipewire code.

**Comment by LebedevRI** [comment]
"No audio in one of the channels" is still happening with
```
ii  wireplumber                                      0.4.13-1                                                     amd64        modular session / policy manager for PipeWire
```
Will get the debug logs next time ithappens...

**Comment by wwmm** [comment]
>No audio in one of the channels" is still happening with

So as suspected by one of PipeWire developers the wireplumber is fixing another problem. That is unfortunate but it makes some sense. I think that in the wireplumber issue the player was not being liked to the sink. This is not what happens here. All the links are made and are active. But somehow there is no sound.

**Comment by LebedevRI** [comment]
Observation: this seems to reliable happen when there is an audio playback (a chrome tab with youtube),
that you pause, don't close the tab, and resume some time later, like 10+ minutes later.

**Comment by LebedevRI** [comment]
[pw-dump.txt](https://github.com/wwmm/easyeffects/files/10241492/pw-dump.txt)
[pw.dot.txt](https://github.com/wwmm/easyeffects/files/10241493/pw.dot.txt)

**Comment by wwmm** [comment]
> [pw-dump.txt](https://github.com/wwmm/easyeffects/files/10241492/pw-dump.txt) [pw.dot.txt](https://github.com/wwmm/easyeffects/files/10241493/pw.dot.txt)

As usual all the links are there and according to pw-dot in the active state. Nothing seems to have changed =/.

**Comment by DanMan** [comment]
> Observation: this seems to reliable happen when there is an audio playback (a chrome tab with youtube), that you pause, don't close the tab, and resume some time later, like 10+ minutes later.

It's more like 10s for me. Pause video, wait maybe 10-15s until the stream disappears from the EE outputs tab, resume video: video's frozen again and the YT buffering animation starts playing. The stream in EE came up inactive again and re-enabling it makes the video run again.

**Comment by wwmm** [comment]
I think that this problem got worse in the latest piepewire release because after months without seeing it now I notice it frequently while playing Monster Hunter Rise. In one of the occasions I saw this in pw-dot output
![Screenshot from 2022-12-30 11-05-32](https://user-images.githubusercontent.com/16225172/210088444-478997cc-9618-4b8f-8a3c-e9fdfacff0d8.png)
Some links were stuck in the `paused` state while others were in the `init` state. I attached this image in the issue I have opened at PipeWire's page. If this will give them any idea about what may be going on it is hard to say. But it is clear to me something is going really bad inside PipeWire when links are made "in a bad moment".

**Comment by LebedevRI** [comment]
I think i haven't encountered this issue in last few weeks;
while it was quite random, it used to be somewhat frequent.
(pipewire 0.3.65-2, wireplumber 0.4.13-1, easyeffects 7.0.0-1)

**Comment by LebedevRI** [comment]
Another month flew by. No new occurrences.
I think it might be somehow gone for the moment.

**Comment by wwmm** [comment]
>Another month flew by. No new occurrences.
>I think it might be somehow gone for the moment.

I also haven't seen it for a long time. But I will keep monitoring it for more time.

**Comment by delfianto** [comment]
> I think that this problem got worse in the latest pipewire release because after months without seeing it now I notice it frequently while playing Monster Hunter Rise. In one of the occasions I saw this in pw-dot output ![Screenshot from 2022-12-30 11-05-32](https://user-images.githubusercontent.com/16225172/210088444-478997cc-9618-4b8f-8a3c-e9fdfacff0d8.png) Some links were stuck in the `paused` state while others were in the `init` state. I attached this image in the issue I have opened at PipeWire's page. If this will give them any idea about what may be going on it is hard to say. But it is clear to me something is going really wrong inside PipeWire when links are made "in a bad moment".

Btw I just experience this in the past few days, fresh EndeavourOS installation
- pipewire 0.3.67
- wireplumber 0.4.14
- easyeffects 7.0.1

Sometimes the links are stuck in init state after simply playing a video in youtube (google chrome) and pausing or stopping it. The occurrence is really random and still have no idea on what is wrong.

Restarting easyeffects with `/usr/bin/easyeffects --gapplication-service` temporarily alleviates the issue but it will happen again after sometime.

**Comment by wwmm** [comment]
>Sometimes the links are stuck in init state after simply playing a video in youtube (google chrome) and pausing or stopping it. The occurrence is really random and still have no idea on what is wrong.

So the bug is just harder to reproduce on some setups. I guess things are getting better but PipeWire definitely still has some weird link management problem.

**Comment by lilydjwg** [comment]
I can still reproduce this (if what happened to me is indeed this issue).

**Comment by wwmm** [comment]
I think we can close this issue. Similar problems probably have a different origin. The corresponding issue at PipeWire's page have also been closed.

**Comment by pinkflames** [comment]
@lilydjwg : are you sure it's the same issue and that you have a fixed EasyEffects version? Please note that this almost exclusively and most obviously affects the right channel, because it was processed later. If you always lose all channels, that may be a different issue and not even related to EasyEffects at all (or EE could only be acting as the trigger due to extra processing and possibly latency).

**Comment by DanMan** [comment]
FWIW: I haven't had this/my issue in a while now.

**Comment by lilydjwg** [comment]
@pinkflames oh, I lost both channels so that should be another issue, and recently I'm not paying a close eye to EE's issues but instead try to restart EE, reconnect the speaker, or pause and play. At least sometimes it's the speaker's fault.

**Comment by pinkflames** [comment]
To be clear, it's not impossible for the issue to affect both channels sometimes. It's just that more often it only broke the right channel, because the link for it would get activated after the left channel's link and there were race conditions around that.
